RFC/0000-gateway-redirects.md

```diff
@@ -0,0 +1,102 @@
+# RFC 0000: Gateway Redirects
+
+- Start Date: (format: 2022-06-15)
+- Related Issues:
+  - https://github.com/ipfs/specs/issues/257
+  - https://github.com/ipfs/go-ipfs/pull/8890
+
+# Summary
+
+Provide support for URL redirects and rewrites for web sites hosted on Subdomain or DNSLink Gateways, thus enabling support for single-page applications (SPAs).
+
+# Motivation
+
+Web sites often need to redirect from one URL to another, for example, to change the appearance of a URL, to change where content is located without changing the URL, to redirect invalid URLs to a pretty 404 page, or to enable URL rewriting. URL rewriting in particular is a critical feature for hosting SPAs, allowing routing logic to be handled by front end code. SPA support is the primary impetus for this RFC.
+
+Currently the only way to handle URL redirects or rewrites is with additional software such as NGINX sitting in front of the Gateway. This software introduces operational complexity and decreases the uniformity of experience when navigating to content hosted on a Gateway, thus decreasing the value proposition of hosting web sites in IPFS.
+
+This RFC proposes the introduction of redirect support for content hosted on Subdomain or DNSLink Gateways, configured via a `_redirects` file residing underneath the root CID of the web site.
+
+# Detailed design
+
+Allow developers to configure redirect support by adding redirect rules to a file named `_redirects` stored underneath the root CID of their web site. The format for this file is similar to those of [Netlify](https://docs.netlify.com/routing/redirects/#syntax-for-the-redirects-file) and [Cloudflare Pages](https://developers.cloudflare.com/pages/platform/redirects) but only supporting a subset of their functionality.
+
+The format for the file is `from to [status]`.
+
+- `from` - specifies the path to intercept (can include placeholders and a trailing splat)
+- `to` - specifies the path or URL to redirect to (can include placeholders or splat matched in `from`)
+- `status` - optional HTTP status code (301 if not specified)
+
+Rules in the file are evaluated top to bottom.
+
+For performance reasons this proposal does not include forced redirect support (i.e. redirect rules that are evaluated even if the `from` path exists). In other word, redirect logic will be evaluated if and only if the requested path does not exist.  If the requested path exists, we won't even check for the existence of the `_redirects` file.
+
+If a `_redirects` file exists but is unable to be processed, perhaps not even parsing correctly, errors will be returned to the user viewing the site via the Gateway.
+
+## Test fixtures
```

Comment:
Move this section to `REDIRECTS_FILE.md` and only list CID that should be pinned in IPIP document.

Reply:
Done, but will await further feedback. 👍